### DISCUSSION TITLE
Performance issue with Chrome >98 and MacOS+Radeon 5xx GPU

### DISCUSSION THREAD

**Discussion by wisdomG** [visgl/deck.gl#6627] [General]
I just found deck.gl didn't work well after I upgrade the Chrome to the version 98.0.4758.80. Even all the offical examples (https://deck.gl/examples) can not be used as normal
Dose anyone meet this situation ?

**Comment by tnederlof** [comment]
I am having the same issue with 98.0.4758.102 on MacOSX

**Comment by ibgreen** [comment]
Thanks for reporting. As far as we know, the majority users do not experience this which can mean that this is specific to e.g. certain GPUs / drivers / os versions. If reporters can provide more information. E.g. Model/year/screen size/GPU of your MacBook.

**Comment by ibgreen** [comment]
Also Chrome 99 is out, and you can also install Chrome Canary (currently on v101) and see if the problem persists or if the Chrome team has already found it and fixed it.

**Comment by ryanrdetzel** [comment]
I tried this page https://deck.gl/examples/point-cloud-layer/ on the latest chrome and the canary (102.0.4971.0) and it doesn't work. That page loads fine in Firefox though.

My system stats. 
Model Name:	MacBook Pro
Model Identifier:	MacBookPro15,1
Processor Name:	6-Core Intel Core i9
Processor Speed:	2.9 GHz
Number of Processors:	1
Total Number of Cores:	6
L2 Cache (per Core):	256 KB
L3 Cache:	12 MB
Hyper-Threading Technology:	Enabled
Memory:	32 GB
System Firmware Version:	1731.100.130.0.0 (iBridge: 19.16.14242.0.0,0)
OS Loader Version:	540.100.7~14

Radeon Pro 560X
Type:	GPU
Bus:	PCIe
PCIe Lane Width:	x8
VRAM (Total):	4 GB
Vendor:	AMD (0x1002)
Device ID:	0x67ef
Revision ID:	0x00c2
ROM Revision:	113-C980AL-075
VBIOS Version:	113-C97501U-005
EFI Driver Version:	01.A1.075
Automatic Graphics Switching:	Supported
gMux Version:	5.0.0

**Reply by ibgreen** [reply]
This is consistent with independent reports of extreme perf issues on 15" 2019 MacBook Pros with the AMD Radeon 5xx series. Was caused by MacOS upgrades in our case.

So far seems specific to the AMD Radeon 5xx GPU and the Chrome browser.

- the higher end GPU option on the MacBook Pro 2019 model does not seem to exhibit the problem
- neither does the 16" model from that year. 
- in addition, the problem is not present on Firefox or Safari.

Do you have access to other machines so you can confirm whether the issue is specific to this model , GPU?

The fact that this only happens on a specific GPU makes it very tricky to debug.

**Reply by Pessimistress** [reply]
Reported to chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1311291

**Reply by ibgreen** [reply]
Great. FWIW, in our case this happens even without external screens attached.

**Reply by ryanrdetzel** [reply]
TIL; Different graphics chips are used when an external monitor is connected. 
When I disconnect the external monitor it runs fine which is a good sign and a stronger case against the Radeon 🤷

**Reply by ibgreen** [reply]
Perhaps if you start from a clean slate, killing the browser between runs. Once an application (or browser tab) starts using one GPU it is not going to switch GPU until all apps or tabs have released it. 

Specifying `powerPreference 'high-performance'` option in the context creation parameters in example should force the use of the Radeon.

**Reply by emilyxxie** [reply]
To chip in with an additional data point as I work with @ryanrdetzel, it runs perfectly fine on my machine with the new Chrome version and I have the following specs:

**MacOs Big Sur 11.6.5**
MacBook Pro (16-inch, 2019)
Processor: 2.4 GHz 8-Core Intel Core i9
Memory: 64 GB 2667 MHz DDR4
Graphics: AMD Radeon Pro 5500M 8 GB
Intel UHD Graphics 630 1536 MB

**External Monitor:**
Dell UltraSharp 27" 16:9 HDR 4K IPS Monitor

**Reply by ibgreen** [reply]
Update - I am being told that GPU switching for running apps can indeed happen in the external monitor connect/disconnect use case. 

That is still consistent with my belief that the issue is with the Radeon 500 series GPU. It is the switch to the discrete Radeon GPU that causes the performance drop, not the external monitor.

**Reply by ibgreen** [reply]
@emilyxxie Thanks for the data point. Yes our tests show the same, the 16" model has a different GPU series and is not affected.

**Comment by Pessimistress** [comment]
To save your trip down the rabbit hole, this is a regression in Chrome due to some unknown performance bug in Apple's OpenGL drivers. The official recommendation is to turn on Metal backed ANGLE in your Chrome's about:flags. See: https://bugs.chromium.org/p/chromium/issues/detail?id=1245448#c22

**Reply by ibgreen** [reply]
Interesting. Looks like that is available in Chrome Canary only at this time?

**Comment by danielwhatmuff** [comment]
Just to bump this - I still have this issue when using deck.gl in both chrome and chromium based edge. Working fine on firefox.
<img width="255" alt="Screenshot 2022-11-02 at 13 14 01" src="https://user-images.githubusercontent.com/6910521/199498807-87f50f41-9e57-458c-a9d0-2b9a194a632a.png">

**Reply by Pessimistress** [reply]
If you click on the Chromium bug tracker in my other comment you can follow the progress. This is a problem in Chrome (who points their finger at MacOS) and their suggestion is to wait for their new metal-backed renderer to ship.
